Assets/Scripts/FillSystem.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FillSystem : MonoBehaviour
+{
+    public float maxFill;
+    public float currentFill;
+
+    public FillBarUI FillBarUI;
+
+    /* When the game starts set the currentMana to the max it can be
+     * 
+     */
+    void Start()
+    {
+        FillBarUI = gameObject.GetComponent(typeof(FillBarUI)) as FillBarUI;
+        currentFill = maxFill;
+    }
+
+    /* A method which returns the currentFill
+     * 
+     */
+    public float GetCurrentFill()
+    {
+        return currentFill;
+    }
+
+
+    /* This method allows the player to gain mana when called elsewhere
+     * 
+     * If valueGained is negative:
+     *      The programer is told via Debug.LogError that they likely intended to use Consume()
+     * 
+     * Otherwise:
+     *      Set a temporary variable for the currentFill that is used in the ChangeFillAmount() Lerp function
+     *      Set currentFill to the value of currentFill plus valueGained
+     *      Clamp this value to the maxFill so that the player never has more than maxFill
+     *      ChangeFillAmount() based on the new fill level
+     *      
+     */
+    public void Gain(float valueGained)
+    {
+        if (valueGained < 0)
+        {
+            Debug.LogError("The Gain() function expects a positive valueGained not a negative one");
+            Debug.LogError("Did you intend to use the Consume() function?");
```

Comment:
might be better to use a new line char '\n' instead of printing twice. It looks like two errors when its technically one error.  That's not important though just wanted to mention the idea